src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -969,7 +969,7 @@ CHIP_ERROR P256Keypair::Serialize(P256SerializedKeypair & output)
 
     {
         size_t len = output.Length() == 0 ? output.Capacity() : output.Length();
-        BufBound bbuf(output, len);
+        Encoding::LittleEndian::BufferWriter bbuf(output, len);
```

Comment:
There was some discussion in #4014, but I didn't read any consensus from it.

There's also the free functions in `<core/CHIPEncoding.h>`, with similar but not identical scoping, and `Get`/`Put` that don't increment the buffer pointer like the `BufferReader`/`BufferWriter` versions (and `Read`/`Write` that do), all wrapping similar `nl::` functions.

Reply:
This wasn't really raised in #4014.  The discussion there was about having a single writer that can do either endianness.  My suggestion is about having a thing that does "write as long as you run out of buffer, measure how much buffer you need" but does not provide any endianness anything, so you're limited to writing single bytes and arrays of bytes.  The idea would be that if and when you try to write something that has endianness requirements, _then_ you would actually think about which endianness to use.  That seems like it would be a better fit for this particular consumer.

Reply:
Yes; my point was that the single writer would make a non-endian subclass unnecessary.

Either way, a followup, keeping this PR a straight 1:1 replacement of `BufBound`.